testui/support/appium_driver.py

```diff
@@ -420,9 +421,16 @@ def __local_run(url, desired_caps, use_port, udid, log_file):
             file_path = os.path.join(log_dir, f"testui-{udid}-{time.time()}-" + log_file)
         else:
             file_path = os.path.join(log_dir, log_file)
+        # Check Appium Version
+        command = ["appium", "-p", port.__str__(), "-bp", bport.__str__()]
```

Comment:
From what I gather, the `-bp` argument was used only for UiAutomator1, which is no longer supported in Appium 2. I would not expect this framework to be used with UiAutomator1 anyway, so I would suggest removing this, and adapting the code like for the iOS method.

Reply:
In theory it was suggested to be used when running parallel execution and using different appium servers per each device. I think it might work regardless of that, as you pointed out. So not sure if I should remove it

Reply:
In my framework I've had cases with 2 parallel executions, each on a different device and Appium server, and they worked fine - I've never used this `-bp` argument. The key part is to set different Appium servers, as well as unique `systemPort`/`wdaLocalPort` values for each device.